Clusters Updates

BboyAkers opened the pull request.

BboyAkers committed:
delete cluster finished

BboyAkers committed:
only show view if cluster is running, clone ready, or updated

BboyAkers committed:
pushing cluster list auth code

dawsontoth committed:
Group clusters visually by status

dawsontoth committed:
Refetch cluster page when looking at its list

dawsontoth committed:
Place clone ready in yellow status

https://harperdb.atlassian.net/browse/STUDIO-238

dawsontoth committed:
Minor form field corrections

dawsontoth committed:
Shuffle url libs

https://harperdb.atlassian.net/browse/STUDIO-174

dawsontoth committed:
Always await the promise in a loader

https://harperdb.atlassian.net/browse/STUDIO-174

dawsontoth committed:
Cluster level management

https://harperdb.atlassian.net/browse/STUDIO-174

dawsontoth commented:
@BboyAkers I rebased this, and then I implemented these bullet points:
- We'll group the clusters into "Running" and "Inactive".
- The cluster page will refetch every 10 seconds when you're looking at the page.
- If a cluster isn't self-managed, display a ‘Login’ or ‘View’ button. ‘View’ will take us directly to the instance browse page for that cluster with a simplified URL route (no `instance/{id}`).
- If it’s a self-managed cluster hide the ‘Login’ button and display only the ‘Instances’ button, which will take you to the list of instances for the cluster.
- Non-self-managed clusters have a new "Instances" context menu option to take you to the overall list.
- "Clone Ready" is now in the yellow/warning bucket.
- All the instance pages are now routed in a secondary way, when running in the cloud, a la `http://localhost:5173/#/orgs/{orgId}/clusters/{clusterId}/config`, which knows what to do to use the cluster's fqdn.

dawsontoth commented on src/features/cluster/queries/getClusterInfoQuery.ts:
```diff
@@ -3,18 +3,17 @@ import { queryKeys } from '@/react-query/constants';
 	return queryOptions({
 		queryKey: [queryKeys.cluster, clusterId],
-		queryFn: getClusterInfo,
+		queryFn: () => getClusterInfo(clusterId!),
```
I keep going back and forth on this pattern... on the one hand, I really like not creating more contexts. By using the passed `context.queryKey`, memory management becomes simpler for the runtime. But as soon as we go to utilize `getClusterInfo` from somewhere else in the codebase, it's much harder to call...

BboyAkers replied:
Hmmmmm 🤔, while memory management becomes simpler for the runtime, does the overhead of creating more context outweigh the simplicity of being able to utilize this elsewhere in the codebase? I personally say no but it's definitely something for us to be aware of as we improve the standardize patterns in the codebase overtime

dawsontoth commented on src/features/cluster/routes.ts:
```diff
@@ -3,25 +3,41 @@ import { clustersLayoutRoute } from '@/features/clusters/routes';
 	getParentRoute: () => clusterLayoutRoute,
 	path: '/',
 	component: ClusterIndex,
+	// TODO: We're going to want to resolve auth by instanceId and clusterId too for faster checks...
```
🤔 I'm still pondering this one.

dawsontoth committed:
Remove unused cluster field on instance

Since I shifted us to load the instance through the cluster itself (moving us away from the /HDBInstance/{id} route), we won’t get the cluster back nested on the instance.

dawsontoth committed:
Add unit tests for group by

dawsontoth committed:
Appease sonar slightly

dawsontoth committed:
Cluster log out + simplify auth store interactions

dawsontoth reviewed (approved):
Proxy approval 😆
